mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/mail/MemoryMessageWithAttachmentMapperTest.java

```diff
@@ -21,8 +21,15 @@
 
 import org.apache.james.mailbox.store.mail.model.MapperProvider;
 import org.apache.james.mailbox.store.mail.model.MessageWithAttachmentMapperTest;
+import org.junit.Before;
 
 public class MemoryMessageWithAttachmentMapperTest extends MessageWithAttachmentMapperTest {
+    
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
```

Comment:
why don't you rely anymore on abstract class annotations?

Reply:
Good question, the answer is : because it's bad.
Let me explain : the abstract class is here to hold test cases. It allows to share an interface contract test suite reusable between implementation.
The test class is here to provide the test environment : launch the middleware, create the sut, start things.
That later point is "llifecycle". It's why I think it doesn't make sense the "orchestrate" the lifecycle from the abstract class : it's not its purpose.
I strongly believe it's test class responsibility to manage that.
Side note : that unfortunately add some boiterplate. It's why junit 5 has interface tests : you can then implement several of them with a single test class.